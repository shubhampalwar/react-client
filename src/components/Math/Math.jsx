import { Component } from 'react';
import PropTypes from 'prop-types';

class Math extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  calculations = () => {
    const { first, second, operator } = this.props;
    switch (operator) {
    case '+': return first + second;
    case '-': return first - second;
    case '*': return first * second;
    case '/': return (second === 0) ? 'infinity' : first / second;
    default: return 'Invalid Operator';
    }
  }

  render() {
    const {
      first, second, operator, children,
    } = this.props;
    return (children) ? children(
      first, second, operator, this.calculations(),
    ) : `${first} ${operator} ${second} = ${this.calculations()}`;
  }
}

Math.propTypes = {
  first: PropTypes.number.isRequired,
  second: PropTypes.number.isRequired,
  operator: PropTypes.string.isRequired,
  children: PropTypes.func,
};
Math.defaultProps = {
  children: undefined,
};
export default Math;
