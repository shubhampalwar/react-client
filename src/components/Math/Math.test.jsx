import { shallow } from 'enzyme';
import React from 'react';
import Math from './Math';

const wrapper = shallow(<Math first="3" second="3" operator="*" />);

describe('Math', () => {
  it('Should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
