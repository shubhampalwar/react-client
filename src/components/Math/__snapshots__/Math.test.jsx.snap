// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Math Should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Math
    first="3"
    operator="*"
    second="3"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): "3 * 3 = 9",
  Symbol(enzyme.__nodes__): Array [
    "3 * 3 = 9",
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
